### ISSUE TITLE
Relational data is no longer being fetched even when using the 'include' key word.

### ISSUE THREAD

**Issue by nak411** [ParsePlatform/Parse-SDK-Android#329]
My existing applications are breaking because there seems to be some kind of server update where relational data is not being fetched for one to many relations, instead only the object ids are fetched.  I have created a stack overflow post with an example:

http://stackoverflow.com/questions/34599580/why-is-relational-query-not-returning-relational-data

**Comment by wangmengyan95** [comment]
Hi @nak411, I test with your sample code and it works fine in my emulator. In your case, could you
1. Try our [REST api](https://parse.com/docs/rest/guide#queries) with your app to see the return value from Parse Server. The results should be 
```json
{"books":[{"__type":"Object","className":"Book","createdAt":"2016-01-05T18:30:18.714Z","name":"123","objectId":"vO9pTOOlTH","updatedAt":"2016-01-05T18:30:18.714Z"}],"createdAt":"2016-01-05T18:30:18.853Z","objectId":"vrUCLm5vA6","updatedAt":"2016-01-05T18:30:18.853Z"}
```
Note the `__type` of the nested book should be Object. If you see this result, that means or backend returns the right value to you.
2. Try [Parse-Interceptor](https://github.com/ParsePlatform/ParseInterceptors-Android) and paste your request's payload. If the payload is the same with the REST api, that means our SDK works fine.
3. Provide us with your test eovironment, SDK version.
Thanks.

**Comment by nak411** [comment]
My problem is that I am not even getting that object back it is just returning the ids.   Here is the output from network interceptor:

01-05 14:53:14.649 12282-12645/com.test I/ParseLogInterceptor: --------------
01-05 14:53:15.172 12282-12663/com.test I/ParseLogInterceptor: Type : Response
01-05 14:53:15.172 12282-12663/com.test I/ParseLogInterceptor: Request-Id : 1
01-05 14:53:15.172 12282-12663/com.test I/ParseLogInterceptor: Status-Code : 200
01-05 14:53:15.172 12282-12663/com.test I/ParseLogInterceptor: Reason-Phrase : OK
01-05 14:53:15.172 12282-12663/com.test I/ParseLogInterceptor: Headers : {X-Runtime=0.031569, Access-Control-Allow-Methods=*, X-Android-Response-Source=NETWORK 200, Connection=keep-alive, X-Parse-Platform=G1, X-Android-Sent-Millis=1452023594666, Content-Type=application/json; charset=utf-8, Date=Tue, 05 Jan 2016 19:53:09 GMT, Server=nginx/1.6.0, Content-Length=193, X-Android-Received-Millis=1452023595171, Access-Control-Allow-Origin=*}
01-05 14:53:15.173 12282-12663/com.test I/ParseLogInterceptor: Body : {
                                                                                         "results": [
                                                                                             {
                                                                                                 "books": [
                                                                                                     {
                                                                                                         "objectId": "aXupljJUw7"
                                                                                                     },
                                                                                                     {
                                                                                                         "objectId": "I5iLHfTDot"
                                                                                                     }
                                                                                                 ],
                                                                                                 "createdAt": "2016-01-05T19:32:10.347Z",
                                                                                                 "name": "Foo",
                                                                                                 "objectId": "IGpgm3y1O0",
                                                                                                 "updatedAt": "2016-01-05T19:33:38.055Z"
                                                                                             }
                                                                                         ]
                                                                                     }
01-05 14:53:15.173 12282-12663/com.test I/ParseLogInterceptor: --------------

I am also including the author and book class from the data browser so you can see that the data is there:

![screen shot 2016-01-05 at 2 57 46 pm](https://cloud.githubusercontent.com/assets/4589745/12126035/e003dfd0-b3bc-11e5-8332-f8627424b0e6.png)
![screen shot 2016-01-05 at 2 58 01 pm](https://cloud.githubusercontent.com/assets/4589745/12126037/e19fccdc-b3bc-11e5-8c4e-a89d832d5cc5.png)

**Comment by nak411** [comment]
If it  helps, I can produce it more consistently by having multiple data rows.  Here is the response, notice the first object only returns ids while the others return the full object even though all items are exactly the same.

 I/ParseLogInterceptor: 

    {
    "results": [
      {
        "books": [
          {
            "objectId": "u6SDSJZYlo"
          },
          {
            "objectId": "fluiqFH1W9"
          }
        ],
        "createdAt": "2016-01-05T21:03:29.334Z",
        "name": "Foo",
        "objectId": "DNdYwn0tZy",
        "updatedAt": "2016-01-05T21:03:59.147Z"
      },
      {
        "books": [
          {
            "__type": "Object",
            "className": "Book",
            "createdAt": "2016-01-05T21:04:25.701Z",
            "objectId": "1As8CFXglI",
            "title": "Title 1",
            "updatedAt": "2016-01-05T21:04:25.701Z"
          },
          {
            "__type": "Object",
            "className": "Book",
            "createdAt": "2016-01-05T21:04:25.701Z",
            "objectId": "qw6ieZlzii",
            "title": "Title 2",
            "updatedAt": "2016-01-05T21:04:25.701Z"
          }
        ],
        "createdAt": "2016-01-05T21:04:25.795Z",
        "name": "Foo",
        "objectId": "I6QW79eX8u",
        "updatedAt": "2016-01-05T21:04:25.795Z"
      },
      {
        "books": [
          {
            "__type": "Object",
            "className": "Book",
            "createdAt": "2016-01-05T21:04:33.943Z",
            "objectId": "J7jDqG7jkL",
            "title": "Title 1",
            "updatedAt": "2016-01-05T21:04:33.943Z"
          },
          {
            "__type": "Object",
            "className": "Book",
            "createdAt": "2016-01-05T21:04:33.943Z",
            "objectId": "PWGyz6h7vS",
            "title": "Title 2",
            "updatedAt": "2016-01-05T21:04:33.943Z"
          }
        ],
        "createdAt": "2016-01-05T21:04:34.773Z",
        "name": "Foo",
        "objectId": "Vr8hxl40Wy",
        "updatedAt": "2016-01-05T21:04:34.773Z"
      }
     ]
     }

**Comment by wangmengyan95** [comment]
Hi @nak411, it is really weird, could you try our REST API to see what happens?

**Comment by nak411** [comment]
Hello @wangmengyan95 I tested with your rest api but received the same response.  However I did figure out the problem. It was with how the data is saved.  It seems like save in background wasn't properly creating relations but the problem seems to be on the server side since it is now working today.

Note: I have made 0 changes to the test app.

The one difference I have noticed is that the relation column now __type and className field which it did not before even though same code/testing environment is being used to save the objects.

This is what it was looking like before in the data browser when  it wasn't working (notice that books column)

![before](https://cloud.githubusercontent.com/assets/4589745/12155770/01170ee4-b495-11e5-99ee-3395b1bd60a5.png)


and this is what it looks like now

![after](https://cloud.githubusercontent.com/assets/4589745/12155784/0df4a90a-b495-11e5-870b-515c2b1b2f52.png)

I'm not sure what was wrong but the problem seems to be with the server side since no client code was changed.  I am closing this issue for now.
